### ISSUE TITLE
[feature request] Remove external mysql requisite

### ISSUE THREAD

**Issue by Tom-Neverwinter** [SapphireServer/Sapphire#526]
**Describe the problem:** *Please describe the problem you're encountering accurately, including steps to reproduce/recreate the issue.*
Mysql sucks. I use mariadb to completely ignore the agony of using this software. (updating, connection, failure to have a lot of concurrent connections, uninstall, bloat, etc.)

**Screenshots or videos needed to identify the problem:** *Please attach screenshots or videos showing the problem to this issue, if applicable.*
N/A

**Sapphire branch and commit:** *You can get this by checking the server startup message in your console window, running `` git describe --all`` in a console window or looking at your git client.*
N/A

**Logs:** *You can get these from the /bin/logs directory in your sapphire clone, please attach them to this issue.*
N/A
**Setup:** *Please note down which operating system you are using and any other information about your setup which could be of use to us, like compiler/Visual Studio version and FFXIV version.*
N/A

**Comment by NotAdam** [comment]
I assume you're on windows?

**Comment by Tom-Neverwinter** [comment]
yup. mysql is already trying to cause hell.

**Comment by NotAdam** [comment]
This'll be because mariadb gets installed to a different location, one we don't search for in `FindMySQL.cmake`. is the path to your mariadb install `C:\Program Files\MariaDB 10.3`?

**Comment by Tom-Neverwinter** [comment]
I just deleted it after vs found what it needed for its install part. mysql isnt needed. just the relevant vs parts, as I suspected.

mariadb is running the show now just fine and cmake can do its thing.

**Comment by NotAdam** [comment]
if you delete findmysql from the build process, common lib won't build anymore. It is absolutely needed, you probably just haven't noticed it because the cmake integration in VS is absolutely awful. I'll fix this properly in a bit

**Comment by NotAdam** [comment]
should be fixed for you on develop now at least, let me know how you go with that

**Comment by Tom-Neverwinter** [comment]
```
Severity	Code	Description	Project	File	Line	Suppression State
Error		CMake Error at cmake/FindMySQL.cmake:135 (file):
  file COPY cannot find "C:/Program Files/MySQL/MySQL Server
  5.7/lib/libmysql.dll".		C:\Sapphire\cmake/FindMySQL.cmake	135	

```
after nuking it just in case it still throws a fit

**Comment by NotAdam** [comment]
Did you install mysql dev libs alongside mariadb?

**Comment by NotAdam** [comment]
erm, as in did you install mysql separately alongside mariadb? does libmysql.dll even exist in `C:/Program Files/MySQL/MySQL Server 5.7/lib/libmysql.dll`?

**Comment by Tom-Neverwinter** [comment]
~~I think ~~ I uninstalled it in the nuke.

**Comment by NotAdam** [comment]
and it's still using the path as above?

**Comment by Tom-Neverwinter** [comment]
yes, I am going to experiment with the mariadb-connector-c and see if I can bypass reinstalling mysql (also LGPL so it can be included with its own license in the build)

**Comment by NotAdam** [comment]
mariadb includes the mysql library as is (but has some extra stuff too) and works without changes on my linux vm - it won't need any code changes to work. Likely cmake is picking up things that still (partially) exist and then throwing a fit because it can't find the rest

**Comment by NotAdam** [comment]
ah nevermind, I see what the issue is, mariadb client on windows is called libmariadb instead of libmysql

**Comment by Tom-Neverwinter** [comment]
I must have renamed it last night. stayed up way too late after I got it working XD maybe 8am cst

**Comment by NotAdam** [comment]
can you try this?
```diff
diff --git a/cmake/FindMySQL.cmake b/cmake/FindMySQL.cmake
index d60044c3..8413f8ce 100644
--- a/cmake/FindMySQL.cmake
+++ b/cmake/FindMySQL.cmake
@@ -113,6 +113,7 @@ if( WIN32 )
   find_library( MYSQL_LIBRARY
     NAMES
       libmysql
+      libmariadb
     PATHS
       ${MYSQL_ADD_LIBRARIES_PATH}
       "${PROGRAM_FILES_64}/MySQL/MySQL Server 5.7/lib"```

**Comment by Tom-Neverwinter** [comment]
Severity	Code	Description	Project	File	Line	Suppression State
Error		CMake Error at cmake/FindMySQL.cmake:195 (message):
  Could not find MySQL headers! Please install the development libraries and
  headers		C:\Sapphire\cmake/FindMySQL.cmake	195

**Comment by NotAdam** [comment]
alright, add this in too

```diff
diff --git a/cmake/FindMySQL.cmake b/cmake/FindMySQL.cmake
index d60044c3..b43bf6f0 100644
--- a/cmake/FindMySQL.cmake
+++ b/cmake/FindMySQL.cmake
@@ -82,6 +82,7 @@ find_path(MYSQL_INCLUDE_DIR
     /usr/local/mysql/include
     "${PROGRAM_FILES_64}/MySQL/MySQL Server 5.7/include"
     "${PROGRAM_FILES_64}/MySQL/include"
-    "${PROGRAM_FILES_64}/MariaDB 10.3/include"
+    "${PROGRAM_FILES_64}/MariaDB 10.3/include/mysql"
     "C:/MySQL/include"
     "[HKEY_LOCAL_MACHINE\\SOFTWARE\\MySQL AB\\MySQL Server 5.7;Location]/include"
     "$ENV{ProgramFiles}/MySQL/MySQL Server 5.7/include"
```

**Comment by Tom-Neverwinter** [comment]
compiling, compiling. please hold

**Comment by NotAdam** [comment]
does api/world start too?

**Comment by Tom-Neverwinter** [comment]
windows install path: 

C:\Program Files (x86)\SquareEnix\FINAL FANTASY XIV - A Realm Reborn\game\sqpack

C:\\Program Files (x86)\\SquareEnix\\FINAL FANTASY XIV - A Realm Reborn\\game\\sqpack

testing now

**Comment by NotAdam** [comment]
what's the error?

build path for msvc can be adjusted in cmakesettings.json

**Comment by Tom-Neverwinter** [comment]
changed to C:\Sapphire\bin for build. 

I do not have an error out, let me grab a screenshot

**Comment by NotAdam** [comment]
cool, I'll merge those things into findmysql.cmake.

navmeshes are generated with nav_export or alternatively pcb_reader2, completely separate to the db. There's no documentation yet either as it's still somewhat wip

**Comment by Tom-Neverwinter** [comment]
HYPE! that resolves this issue, THANK YOU!

**Comment by NotAdam** [comment]
fixed in #528
